Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM python:3.12.3-bookworm as base
+ENV POETRY_HOME="/opt/poetry" \
+    POETRY_VERSION=1.7.1 
+ENV PATH="$PATH:$POETRY_HOME/bin"
+RUN curl -sSL https://install.python-poetry.org | python3 -
+WORKDIR /code
+COPY poetry.lock pyproject.toml /code/
+RUN poetry install
+COPY ./todo_app /code/todo_app
```

Comment:
Since this is reasonably targeted, a `.dockerignore` isn't critical here, but I'd still recommend it for good practice; you can skip a `.gitignore` if you trust your devs to all consistently remember which files not to copy in but at some point someone _will_ mess up!

OK, that's slightly a unfair comparison since for `.env` for example, the `.dockerignore` won't do anything at all right now, but I'd still heavily encourage it to be defensive and prevent someone in future from changing the COPY or moving the `.env` into the `todo_app` folder and missing the knock on consequences

Right now, another entry that could bite you and is worth docker-ignoring is `**/__pycache__` since cached files can cause frustrating bugs at times!

Reply:
(Admittedly this is partly an excuse to mention this `**/__pycache__` because I don't expect people to necessarily be aware that `.dockerignore` has slightly different behaviour to `.gitignore`, in particular in that it _doesn't_ ignore subfolders by default - we need to explicitly reference them with the `**` prefix to mean arbitrary levels of nesting should be considered for this rule)